In [1]:
import pandas as pd
import numpy as np

In [2]:
df1 = pd.read_csv('/Users/macbookpro/Downloads/airline customer/test.csv')
df2 = pd.read_csv('/Users/macbookpro/Downloads/airline customer/train.csv')
df = pd.concat([df1,df2], axis =0 , ignore_index = True)

In [3]:
# remove non useful cloumns: Unamed, id
df = df.iloc[:,2:]

In [4]:
df

Gender      Customer Type  Age   Type of Travel     Class  \
0       Female     Loyal Customer   52  Business travel       Eco   
1       Female     Loyal Customer   36  Business travel  Business   
2         Male  disloyal Customer   20  Business travel       Eco   
3         Male     Loyal Customer   44  Business travel  Business   
4       Female     Loyal Customer   49  Business travel       Eco   
...        ...                ...  ...              ...       ...   
129875  Female  disloyal Customer   23  Business travel       Eco   
129876    Male     Loyal Customer   49  Business travel  Business   
129877    Male  disloyal Customer   30  Business travel  Business   
129878  Female  disloyal Customer   22  Business travel       Eco   
129879    Male     Loyal Customer   27  Business travel  Business   

        Flight Distance  Inflight wifi service  \
0                   160                      5   
1                  2863                      1   
2                   192                      2   
3                  3377                      0   
4                  1182                      2   
...                 ...                    ...   
129875              192                      2   
129876             2347                      4   
129877             1995                      1   
129878             1000                      1   
129879             1723                      1   

        Departure/Arrival time convenient  Ease of Online booking  \
0                                       4                       3   
1                                       1                       3   
2                                       0                       2   
3                                       0                       0   
4                                       3                       4   
...                                   ...                     ...   
129875                                  1                       2   
129876                                  4                       4   
129877                                  1                       1   
129878                                  1                       1   
129879                                  3                       3   

        Gate location  ...  Inflight entertainment  On-board service  \
0                   4  ...                       5                 5   
1                   1  ...                       4                 4   
2                   4  ...                       2                 4   
3                   2  ...                       1                 1   
4                   3  ...                       2                 2   
...               ...  ...                     ...               ...   
129875              3  ...                       2                 3   
129876              4  ...                       5                 5   
129877              3  ...                       4                 3   
129878              5  ...                       1                 4   
129879              3  ...                       1                 1   

        Leg room service  Baggage handling  Checkin service  Inflight service  \
0                      5                 5                2                 5   
1                      4                 4                3                 4   
2                      1                 3                2                 2   
3                      1                 1                3                 1   
4                      2                 2                4                 2   
...                  ...               ...              ...               ...   
129875                 1                 4                2                 3   
129876                 5                 5                5                 5   
129877                 2                 4                5                 5   
129878                 5                 1                5               

In [5]:
df.shape

(129880, 23)

In [ ]:
# There is 129880 lines and 23 columns in our dataset.

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 129880 entries, 0 to 129879
Data columns (total 23 columns):
 #   Column                             Non-Null Count   Dtype  
---  ------                             --------------   -----  
 0   Gender                             129880 non-null  object 
 1   Customer Type                      129880 non-null  object 
 2   Age                                129880 non-null  int64  
 3   Type of Travel                     129880 non-null  object 
 4   Class                              129880 non-null  object 
 5   Flight Distance                    129880 non-null  int64  
 6   Inflight wifi service              129880 non-null  int64  
 7   Departure/Arrival time convenient  129880 non-null  int64  
 8   Ease of Online booking             129880 non-null  int64  
 9   Gate location                      129880 non-null  int64  
 10  Food and drink                     129880 non-null  int64  
 11  Online boarding                    1298

In [7]:
# check the data where is null
null = df[df.isnull().T.any()]
null

Gender      Customer Type  Age   Type of Travel     Class  \
516     Female     Loyal Customer   21  Personal Travel       Eco   
656       Male     Loyal Customer    9  Personal Travel       Eco   
1071      Male  disloyal Customer   25  Business travel       Eco   
1224      Male     Loyal Customer    7  Personal Travel       Eco   
1589    Female     Loyal Customer   58  Business travel       Eco   
...        ...                ...  ...              ...       ...   
128043    Male     Loyal Customer   49  Personal Travel  Eco Plus   
128360    Male     Loyal Customer   58  Business travel       Eco   
128528  Female  disloyal Customer   29  Business travel       Eco   
128936    Male     Loyal Customer   58  Business travel       Eco   
129516  Female     Loyal Customer   33  Personal Travel       Eco   

        Flight Distance  Inflight wifi service  \
516                 632                      1   
656                 762                      2   
1071               1017                      3   
1224                399                      4   
1589                129                      5   
...                 ...                    ...   
128043             1249                      2   
128360              733                      3   
128528             1107                      2   
128936             1088                      4   
129516              359                      4   

        Departure/Arrival time convenient  Ease of Online booking  \
516                                     5                       1   
656                                     5                       2   
1071                                    4                       4   
1224                                    5                       4   
1589                                    3                       3   
...                                   ...                     ...   
128043                                  5                       2   
128360                                  3                       3   
128528                                  1                       1   
128936                                  4                       1   
129516                                  4                       4   

        Gate location  ...  Inflight entertainment  On-board service  \
516                 4  ...                       2                 4   
656                 2  ...                       4                 2   
1071                5  ...                       5                 2   
1224                4  ...                       4                 4   
1589                3  ...                       5                 5   
...               ...  ...                     ...               ...   
128043              3  ...                       3                 4   
128360              3  ...                       3                 3   
128528              1  ...                       5                 4   
128936              1  ...                       5                 1   
129516              3  ...                       4                 3   

        Leg room service  Baggage handling  Checkin service  Inflight service  \
516                    2                 5                4                 5   
656                    5                 2                5                 2   
1071                   5                 2                3                 2   
1224                   3                 1                1                 4   
1589                   5                 5                3                 5   
...                  ...               ...              ...               ...   
128043                 5                 4                3                 4   
128360                 1                 2                4                 2   
128528                 1                 5                5                 3   
128936                 5                 5                5               

In [8]:
df.dropna(inplace = True)

In [20]:
df = pd.get_dummies(df)
df.drop(['Gender_Male','Customer Type_disloyal Customer','Type of Travel_Personal Travel','satisfaction_neutral or dissatisfied'],axis =1,inplace = True)

In [21]:
df.describe()

Age  Flight Distance  Inflight wifi service  \
count  129487.000000    129487.000000          129487.000000   
mean       39.428761      1190.210662               2.728544   
std        15.117597       997.560954               1.329235   
min         7.000000        31.000000               0.000000   
25%        27.000000       414.000000               2.000000   
50%        40.000000       844.000000               3.000000   
75%        51.000000      1744.000000               4.000000   
max        85.000000      4983.000000               5.000000   

       Departure/Arrival time convenient  Ease of Online booking  \
count                      129487.000000           129487.000000   
mean                            3.057349                2.756786   
std                             1.526787                1.401662   
min                             0.000000                0.000000   
25%                             2.000000                2.000000   
50%                             3.000000                3.000000   
75%                             4.000000                4.000000   
max                             5.000000                5.000000   

       Gate location  Food and drink  Online boarding   Seat comfort  \
count  129487.000000   129487.000000    129487.000000  129487.000000   
mean        2.976909        3.204685         3.252720       3.441589   
std         1.278506        1.329905         1.350651       1.319168   
min         0.000000        0.000000         0.000000       0.000000   
25%         2.000000        2.000000         2.000000       2.000000   
50%         3.000000        3.000000         3.000000       4.000000   
75%         4.000000        4.000000         4.000000       5.000000   
max         5.000000        5.000000         5.000000       5.000000   

       Inflight entertainment  ...    Cleanliness  Departure Delay in Minutes  \
count           129487.000000  ...  129487.000000               129487.000000   
mean                 3.358067  ...       3.286222                   14.643385   
std                  1.334149  ...       1.313624                   37.932867   
min                  0.000000  ...       0.000000                    0.000000   
25%                  2.000000  ...       2.000000                    0.000000   
50%                  4.000000  ...       3.000000                    0.000000   
75%                  4.000000  ...       4.000000                   12.000000   
max                  5.000000  ...       5.000000                 1592.000000   

       Arrival Delay in Minutes  Gender_Female  Customer Type_Loyal Customer  \
count             129487.000000  129487.000000                 129487.000000   
mean                  15.091129       0.507410                      0.816862   
std                   38.465650       0.499947                      0.386781   
min                    0.000000       0.000000                      0.000000   
25%                    0.000000       0.000000                      1.000000   
50%                    0.000000       1.000000                      1.000000   
75%                   13.000000       1.000000                      1.000000   
max                 1584.000000       1.000000                      1.000000   

       Type of Travel_Business travel  Class_Business      Class_Eco  \
count                   129487.000000   129487.000000  129487.000000   
mean                         0.690764        0.478735       0.448825   
std                          0.462180        0.499550       0.497376   
min                          0.000000        0.000000       0.000000   
25%                          0.000000        0.000000       0.000000   
50%                          1.000000        0.000000       0.000000   
75%                          1.000000        1.000000       1.000000   
max                          1.000000        1.000000       1.000000   

       Class_Eco Plus  satisfaction_satisfied  
count   129487.000000         

We can see that the passengers average age is around 40 years old, the youngest is 7, and the oldest is 85. 81.7% of customers are loyal customers,and 69% of travel is business travel, which explains 47.9% of class type is buiness class.

The satisfaction rate is 43.45%.

The average flight distence is 1190Km.

Infligt wifi service, ease of online booking are relatively low compare to others.

The average departure delay is 14.64 minutes. 

The max of departure delay is 1592minutes, which is around 26.5 hours. 